JavaScripts/bacteriaDNA.js

```diff
@@ -0,0 +1,185 @@
+            this.dna[randomIndex] = mutation;
+            return this.dna;
+        },
+        // TASK 3: compare two bacteria's DNA
+        compareDNA(otherBacteria) {
+            // iterate over each base of this.dna
+            // compare with iteration of base of otherBacteria.dna
+            let count = 0;
+            // if equal count up 1
+            // return count divided by 15 (length of dna) as percentage
+            this.dna.forEach((base, index) => {
+                if (base === otherBacteria.dna[index]) {
+                    count++;
+                }
+            });
+            let percentageOverlap = ((count / this.dna.length) * 100).toFixed(2);
+            console.log(`specimen #${this.specimenNum} and specimen #${otherBacteria.specimenNum
+                } have ${percentageOverlap}% DNA in common.`);
+            return percentageOverlap;
+        },
+        // TASK 4: if either C or G accounts for 60% of the DNA will return true
+        willLikelySurvive() {
+            // counts all C and G bases and returns true if either is 60%
+            let countC = 0;
+            let countG = 0;
+            this.dna.forEach((base) => {
+                if (base === "C") {
+                    countC++;
+                } else if (base === "G") {
+                    countG++;
+                }
+            });
+            // console.log(`C: ${countC}, G: ${countG}`);
+            let percentageC = (countC / this.dna.length) * 100;
+            let percentageG = (countG / this.dna.length) * 100;
+            // console.log(`%C: ${percentageC}, %G: ${percentageG}`)
+            if (percentageC >= 60 || percentageG >= 60) {
+                return true;
+            } else {
+                return false;
+            }
+        },
+        // TASK 6: method that creates complementary DNA strands
+        complementStrand() {
+            return this.dna.map(
+                (base) => {
+                    switch (base) {
+                        case "A": return "T"
+                            break;
+                        case "T": return "A"
+                            break;
+                        case "G": return "C"
+                            break;
+                        case "C": return "G"
+                            break;
+                        default:
+                            break;
+                    }
+                }
+            )
+        }
+    };
+}
+
+// TASK 5: creating 30 instances of bacteria DNA
+let pAequorDatabase = [];
+for (let i = 1; i <= 30; i++) {
+    pAequorDatabase.push(pAequorFactory(i, mockUpStrand));
+}
+
+// STILL OPEN! 
+// TASK 7: finding highest matching pair from DNA Database
+// let matches =
+//     pAequorDatabase.map((DNAStrand, index1) => {
+//         DNAStrand.compareDNA(pAequorDatabase
+//             .filter((DNAStrand, index2) => {
+//                 index2 !== index1
+//             }
+//             ))
+//     }
+//     );
```

Comment:
this is the hardest problem

The  algorithm should be:

let bestA, bestB, bestPercentage. 
for each dna in the database:
    for each dna2 in the database again (nested loop):
        if dna === dna2, continue
        percentage = calcPercentage(dna, dna2)
        if (percentage > bestPercentage) {
            bestA = dna;
            bestB = dna2;
            bestPercentage = percentage;
        }